docs/modules/ROOT/pages/node_pattern.adoc

```diff
@@ -47,20 +47,75 @@ def on_send(node)
 end
 ----
 
-== `(` and `)` Navigate deeply with Parens
+== Ruby Abstract Syntax Tree (AST)
 
-Parens delimits navigation inside node and its children.
+Parser translates Ruby source code to a tree structure represented in text.
+A simple integer literal like `1` is represented by `(int 1)` in the AST.
+A method call with two integer literals:
 
-A simple integer like `1` is represented by `(int 1)` in the AST.
+[source,ruby]
+----
+foo(1, 2)
+----
+
+is represented with:
+
+[source]
+----
+(send nil :foo
+  (int 1)
+  (int 2)
+)
+----
+
+Every node is represented with a sequence of children, the first of which is the node type, and others are optionally present and depend on the node type and the parsed source Ruby code.
```

Comment:
Given `#children` does not return the type, I think this should be tweaked. Elements in parenthesis are really the type and then children.

Reply:
Updated.